### PR TITLE
Cpr 282 remove pnc requirement hmcts

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/PersonRepository.kt [hunk 1]
```diff
@@ -14,6 +14,8 @@ interface PersonRepository : JpaRepository<PersonEntity, Long> {
   fun findByDefendantsPncNumber(pncIdentifier: PNCIdentifier): PersonEntity?
   fun findByPrisonersPncNumber(pncIdentifier: PNCIdentifier): PersonEntity?
 
+  fun findByDefendantsDefendantId(defendantId: String): PersonEntity?
```
Comment (vb-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/jpa/repository/PersonRepository.kt:17]: This is a temp method just for the tests.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/matcher/OffenderMatcher.kt [hunk 2]
```diff
@@ -12,12 +12,9 @@ class OffenderMatcher(offenderDetails: List<OffenderDetail>?, person: Person) :
     }
   }
   override fun isPartialMatchItem(item: OffenderDetail): Boolean {
-    return person.otherIdentifiers?.pncIdentifier == PNCIdentifier.from(item.otherIds.pncNumber) &&
-      item.firstName.equals(person.givenName, true) || item.surname.equals(person.familyName, true) || item.dateOfBirth == person.dateOfBirth
+    return item.firstName.equals(person.givenName, true) || item.surname.equals(person.familyName, true) || item.dateOfBirth == person.dateOfBirth
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/matcher/OffenderMatcher.kt:15]: as above, pnc could be used in partial match still

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/matcher/PrisonerMatcher.kt [hunk 3]
```diff
@@ -2,23 +2,15 @@ package uk.gov.justice.digital.hmpps.personrecord.service.matcher
...
 class PrisonerMatcher(prisonerDetails: List<Prisoner>?, person: Person) :
   Matcher<Prisoner>(prisonerDetails, person) {
-  fun isPncDoesNotMatch(): Boolean {
-    return !items.isNullOrEmpty() && items.none {
-      person.otherIdentifiers?.pncIdentifier == PNCIdentifier.from(it.pncNumber)
-    }
-  }
   override fun isPartialMatchItem(item: Prisoner): Boolean {
-    return person.otherIdentifiers?.pncIdentifier == PNCIdentifier.from(item.pncNumber) &&
-      item.firstName.equals(person.givenName, true) || item.lastName.equals(person.familyName, true) || item.dateOfBirth == person.dateOfBirth
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/matcher/PrisonerMatcher.kt:16]: as above

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/type/TelemetryEventType.kt [hunk 4]
```diff
@@ -10,16 +10,12 @@ enum class TelemetryEventType(val eventName: String) {
   HMCTS_PARTIAL_MATCH("CprHMCTSRecordPartialMatchFound"),
   HMCTS_PROCESSING_FAILURE("CprHMCTSMessageProcessingFailed"),
   HMCTS_RECORD_CREATED("CprHMCTSRecordCreated"),
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/type/TelemetryEventType.kt:12]: DELIUS_PNC_MISMATCH is no longer used I think

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/CourtCaseEventsListenerIntTest.kt [hunk 5]
```diff
@@ -436,4 +388,38 @@ class CourtCaseEventsListenerIntTest : IntegrationTestBase() {
...
+    assertThat(personEntity.prisoners[0].dateOfBirth).isEqualTo(LocalDate.of(1970, 3, 15))
+    assertThat(personEntity.prisoners[0].cro).isEqualTo(CROIdentifier.from("51072/62R"))
+    assertThat(personEntity.prisoners[0].fingerprint).isEqualTo(true)
+    assertThat(personEntity.prisoners[0].drivingLicenseNumber).isEqualTo("ERIC1234567K")
+    assertThat(personEntity.prisoners[0].nationalInsuranceNumber).isEqualTo("PD123456D")
+    assertThat(personEntity.prisoners[0].address?.postcode).isEqualTo("LI1 5TH")
+    assertThat(personEntity.prisoners[0].sexCode).isNull()
+    assertThat(personEntity.prisoners[0].raceCode).isNull()
+    assertThat(personEntity.prisoners[0].birthPlace).isNull()
+    assertThat(personEntity.prisoners[0].birthCountryCode).isNull()
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/CourtCaseEventsListenerIntTest.kt:417]: We can probably reduce the number of assertions here - we only really care that the record has been created I think

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/helper/HmctsMessages.kt [hunk 6]
```diff
@@ -618,3 +618,109 @@ fun commonPlatformHearingWithAdditionalFields() = """
       }
     }
 """.trimIndent()
+
+fun commonPlatformHearingWithNewDefendantAndNoPnc() = """
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/helper/HmctsMessages.kt:622]: couldn't this method be replaced by a call to

```commonPlatformHearingWithNewDefendant(pncNumber="")```

**Review by markreesmoj — APPROVED** [review]

**Pull request merged**